src/vs/workbench/api/common/extHostTypes.ts

```diff
@@ -1943,11 +1944,15 @@ export enum DecorationRangeBehavior {
 }
 
 export namespace TextEditorSelectionChangeKind {
-	export function fromValue(s: string | undefined) {
+	export function fromValue(s: TextEditorSelectionSource | string | undefined) {
 		switch (s) {
 			case 'keyboard': return TextEditorSelectionChangeKind.Keyboard;
 			case 'mouse': return TextEditorSelectionChangeKind.Mouse;
-			case 'api': return TextEditorSelectionChangeKind.Command;
+			case 'api':
```

Comment:
@jrieken fyi 'api' seems redundant here as that is `case TextEditorSelectionSource.PROGRAMMATIC`

Found by a warning `esbuild` produces:

```
▲ [WARNING] This case clause will never be evaluated because it duplicates an earlier case clause [duplicate-case]

    out-build/vs/workbench/api/common/extHostTypes.js:1570:17:
      1570 │             case "api" /* TextEditorSelectionSource.PROGRAMMATIC */:
           ╵                  ~~~~~

  The earlier case clause is here:

    out-build/vs/workbench/api/common/extHostTypes.js:1569:17:
      1569 │             case 'api':
```